cmd/textile/buckets.go

```diff
@@ -0,0 +1,385 @@
+    mybuck/bar/two.txt
+    mybuck/bar/baz/three.txt
+
+'textile buckets push foo/bar/baz mybuck':
+    mybuck/baz/three.txt
+
+'textile buckets push foo/bar/baz/three.txt mybuck':
+    mybuck/three.txt
+
+'textile buckets push foo/* foo':
+    foo/one.txt
+    foo/bar/two.txt
+    foo/bar/baz/three.txt
+`,
+	Args: cobra.MinimumNArgs(2),
+	Run: func(c *cobra.Command, args []string) {
+		projectID := configViper.GetString("id")
+		if projectID == "" {
+			cmd.Fatal(errors.New("not a project directory"))
+		}
+
+		var names []string
+		var paths []string
+		bucketPath, args := args[len(args)-1], args[:len(args)-1]
+		for _, a := range args {
+			dir := filepath.Dir(a)
+			err := filepath.Walk(a, func(n string, info os.FileInfo, err error) error {
+				if err != nil {
+					cmd.Fatal(err)
+				}
+				if !info.IsDir() {
+					names = append(names, n)
+					var p string
+					if n == a { // This is a file given as an arg
+						// In this case, the bucket path should not include the directory
+						p = filepath.Join(bucketPath, info.Name())
+					} else { // This is a directory given as an arg, or one of its sub directories
+						// The bucket path should maintain directory structure
+						p = filepath.Join(bucketPath, strings.TrimPrefix(n, dir))
+					}
+					paths = append(paths, p)
+				}
+				return nil
+			})
+			if err != nil {
+				cmd.Fatal(err)
+			}
+		}
+		if len(names) == 0 {
+			cmd.End("No files found")
+		}
+
+		prompt := promptui.Prompt{
+			Label: fmt.Sprintf("Add %d files? Press ENTER to confirm", len(names)),
+			Validate: func(in string) error {
+				return nil
+			},
+		}
+		if _, err := prompt.Run(); err != nil {
+			cmd.End("")
+		}
+
+		for i := range names {
+			addFile(projectID, names[i], paths[i])
+		}
+
+		cmd.Success("Pushed %d files to %s", aurora.White(len(names)).Bold(), aurora.White(bucketPath).Bold())
+	},
+}
+
+// @todo: Support Stdin
+func addFile(projectID, name, filePath string) {
+	file, err := os.Open(name)
+	if err != nil {
+		cmd.Fatal(err)
+	}
+	defer file.Close()
+
+	info, err := file.Stat()
+	if err != nil {
+		cmd.Fatal(err)
+	}
+
+	cmd.Message("Pushing %s to %s", aurora.White(name).Bold(), aurora.White(filePath).Bold())
+
+	bar := pbar.New(int(info.Size()))
+	bar.SetTemplate(pbar.Full)
+	bar.Set(pbar.Bytes, true)
+	bar.Set(pbar.SIBytesPrefix, true)
+	bar.Start()
+	progress := make(chan int64)
+	go func() {
+		for up := range progress {
+			bar.SetCurrent(up)
+		}
+	}()
+
+	ctx, cancel := context.WithTimeout(context.Background(), addFileTimeout)
+	defer cancel()
+	if _, _, err = client.PushBucketPath(
+		ctx,
+		projectID,
+		filePath,
+		file,
+		api.Auth{
+			Token: authViper.GetString("token"),
+		},
+		api.WithPushProgress(progress)); err != nil {
+		cmd.Fatal(err)
+	}
+	bar.Finish()
+}
+
+var pullBucketPathCmd = &cobra.Command{
+	Use:   "pull",
+	Short: "Pull a bucket path",
+	Long: `Pull files and directories from a bucket path. All paths will be created if they don't exist.
```

Comment:
Nit: Clarifying what would happen if that file already exist.

Reply:
Hmm, I need to check what happens actually (and make a note in the doc). Most likely it will currently error. If so, I'll make it overwrite in the meantime... the "force" idea would be good here as well.

Reply:
Sounds good!

Reply:
Ok, currently everything is overwritten locally. I added a note to reflect this.